Mathlib/NumberTheory/ArithmeticFunction.lean

```diff
@@ -754,6 +796,31 @@ theorem eq_iff_eq_on_prime_powers [CommMonoidWithZero R] (f : ArithmeticFunction
   exact Finset.prod_congr rfl fun p hp ↦ h p _ (Nat.prime_of_mem_primeFactors hp)
 #align nat.arithmetic_function.is_multiplicative.eq_iff_eq_on_prime_powers Nat.ArithmeticFunction.IsMultiplicative.eq_iff_eq_on_prime_powers
 
+theorem prodPrimeFactors [CommMonoidWithZero R] (f : ℕ → R) :
+    IsMultiplicative (prodPrimeFactors f) := by
+  rw [iff_ne_zero]
+  constructor
+  · apply prodPrimeFactors_apply
+  intro x y hx hy hxy
+  haveI : NeZero x := ⟨hx⟩
+  haveI : NeZero y := ⟨hy⟩
+  simp_rw [prodPrimeFactors_apply]
+  rw[Nat.primeFactors_mul hx hy, ← Finset.prod_union hxy.disjoint_primeFactors]
+
+theorem prodPrimeFactors_add_of_squarefree [CommSemiring R] {f g : ArithmeticFunction R}
+    (hf : IsMultiplicative f) (hg : IsMultiplicative g) {n : ℕ} (hn : Squarefree n) :
+    ∏ᵖ p ∣ n, (f + g) p = (f * g) n := by
+  rw [prodPrimeFactors_apply_of_ne_zero hn.ne_zero]
+  simp_rw [add_apply (f:=f) (g:=g)]
+  rw [Finset.prod_add, mul_apply, sum_divisorsAntidiagonal (f:= λ x y => f x * g y),
```

Comment:
```suggestion
  rw [Finset.prod_add, mul_apply, sum_divisorsAntidiagonal (f := fun x y => f x * g y),
```
You might also be able to write `(f \. * g \.)` for that lambda expression. (Where `\.` should be a centerdot.)